pandas/tests/indexes/test_category.py

```diff
@@ -365,29 +365,36 @@ def test_astype(self):
         tm.assert_index_equal(result, expected)
 
     def test_reindex_base(self):
-
-        # determined by cat ordering
-        idx = self.create_index()
+        # Determined by cat ordering.
+        idx = CategoricalIndex(list("cab"), categories=list("cab"))
         expected = np.arange(len(idx), dtype=np.intp)
 
         actual = idx.get_indexer(idx)
         tm.assert_numpy_array_equal(expected, actual)
 
-        with tm.assert_raises_regex(ValueError, 'Invalid fill method'):
-            idx.get_indexer(idx, method='invalid')
+        with tm.assert_raises_regex(ValueError, "Invalid fill method"):
+            idx.get_indexer(idx, method="invalid")
 
     def test_reindexing(self):
+        np.random.seed(123456789)
 
         ci = self.create_index()
         oidx = Index(np.array(ci))
 
-        for n in [1, 2, 5, len(ci)]:
+        for n in [1, 2, 5]:
             finder = oidx[np.random.randint(0, len(ci), size=n)]
             expected = oidx.get_indexer_non_unique(finder)[0]
 
             actual = ci.get_indexer(finder)
             tm.assert_numpy_array_equal(expected, actual)
 
+        # see gh-17323
```

Comment:
maybe add a comment that this is an indexer equal to the index

Reply:
Couldn't hurt.  Done.